File: arrays/prob1252.py
# 1252 Cells will odd values in a matrix

# 65%
class Solution(object):
    def oddCells(self, n, m, indices):
        """
        :type n: int
        :type m: int
        :type indices: List[List[int]]
        :rtype: int
        """
        arr = [[0 for _ in range(m)] for _ in range(n)]
        # arr= [[0] *m] *n does not work. shallow copy?
        for r, c in indices:
            for j in range(m):
                arr[r][j] += 1
            for i in range(n):
                arr[i][c] += 1
        
        cnt = 0
        for i in range(n):
            for j in range(m):
                if arr[i][j] & 1:
                    cnt += 1
        return cnt

File: math/prob1252.kt
// 1252 Cells with Odd Values in a Matrix
class Solution {
    fun oddCells(m: Int, n: Int, indices: Array<IntArray>): Int {

        var rmap = mutableMapOf<Int, Int>()
        var cmap = mutableMapOf<Int, Int>()

        for ((r, c) in indices) {
            rmap.put(r, rmap.getOrDefault(r, 0) + 1)
            cmap.put(c, cmap.getOrDefault(c, 0) + 1)
        }

        var ans = 0
        for (r in 0 until m) {
            for (c in 0 until n) {
               var tmp = rmap.getOrDefault(r, 0) + cmap.getOrDefault(c, 0)
               ans += if (tmp % 2 == 1) 1 else 0
            }
        }

        return ans

    }
}
